crates/turbo-tasks-memory/src/scope.rs

```diff
@@ -136,6 +134,13 @@ pub struct TaskScopeState {
     /// All child scopes, when the scope becomes active, child scopes need to
     /// become active too
     children: CountHashSet<TaskScopeId, BuildNoHashHasher<TaskScopeId>>,
+    /// flag if this scope has unfinished tasks
+    has_unfinished_tasks: bool,
+    /// Event that will be notified when all unfinished tasks and children are
+    /// done
+    event: Event,
+    /// All parent scopes
+    parents: CountHashSet<TaskScopeId, BuildNoHashHasher<TaskScopeId>>,
```

Comment:
This is going to increase the size of a single `TaskScope` by 48 bytes. But the task scope state is already a very large structure so might not make much difference.

Reply:
yeah, improving that might be a separate PR.

But there are not that many TaskScopes...